pallets/funding/src/tests/1_application.rs

```diff
@@ -0,0 +1,1321 @@
+			project_metadata.auction_round_allocation_percentage = Percent::from_percent(0);
+
+			inst.mint_plmc_to(default_plmc_balances());
+			let jwt = get_mock_jwt(ISSUER_1, InvestorType::Institutional, generate_did_from_account(ISSUER_1));
+			inst.execute(|| {
+				assert_noop!(
+					Pallet::<TestRuntime>::create_project(RuntimeOrigin::signed(ISSUER_1), jwt, project_metadata),
+					Error::<TestRuntime>::AuctionRoundPercentageError
+				);
+			});
+		}
+
+		#[test]
+		fn target_funding_less_than_1000_usd() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let mut project_metadata = default_project_metadata(inst.get_new_nonce(), ISSUER_1);
+			project_metadata.minimum_price = PriceOf::<TestRuntime>::from_float(1.0);
+			project_metadata.total_allocation_size = 999u128;
+
+			inst.mint_plmc_to(default_plmc_balances());
+			let jwt = get_mock_jwt(ISSUER_1, InvestorType::Institutional, generate_did_from_account(ISSUER_1));
+			inst.execute(|| {
+				assert_noop!(
+					Pallet::<TestRuntime>::create_project(
+						RuntimeOrigin::signed(ISSUER_1),
+						jwt.clone(),
+						project_metadata.clone()
+					),
+					Error::<TestRuntime>::FundingTargetTooLow
+				);
+			});
+
+			project_metadata.minimum_price = PriceOf::<TestRuntime>::from_float(0.00000000001);
+			project_metadata.total_allocation_size = 99999999999999u128;
+			inst.execute(|| {
+				assert_noop!(
+					Pallet::<TestRuntime>::create_project(RuntimeOrigin::signed(ISSUER_1), jwt, project_metadata),
+					Error::<TestRuntime>::FundingTargetTooLow
+				);
+			});
+		}
+	}
+}
+
+#[cfg(test)]
+mod edit_project_extrinsic {
+	use super::*;
+
+	#[cfg(test)]
+	mod success {
+		use super::*;
+		#[test]
+		fn project_id_stays_the_same() {
+			let mut inst = MockInstantiator::new(Some(RefCell::new(new_test_ext())));
+			let mut project_metadata = default_project_metadata(inst.get_new_nonce(), ISSUER_1);
+			inst.mint_plmc_to(default_plmc_balances());
+			let jwt = get_mock_jwt(ISSUER_1, InvestorType::Institutional, generate_did_from_account(ISSUER_1));
+			let project_id = inst.create_new_project(project_metadata.clone(), ISSUER_1);
+
+			project_metadata.minimum_price = PriceOf::<TestRuntime>::from_float(15.0);
+			assert_ok!(inst.execute(|| crate::Pallet::<TestRuntime>::edit_project(
+				RuntimeOrigin::signed(ISSUER_1),
+				jwt.clone(),
+				project_id,
+				project_metadata.clone()
+			)));
+			let next_project_id = inst.execute(|| NextProjectId::<TestRuntime>::get());
+			assert_eq!(project_id, next_project_id - 1);
+			let projects_details = inst.execute(|| ProjectsDetails::<TestRuntime>::iter_keys().collect_vec());
+			let project_metadatas = inst.execute(|| ProjectsMetadata::<TestRuntime>::iter_keys().collect_vec());
+			assert_eq!(projects_details, vec![project_id]);
+			assert_eq!(project_metadatas, vec![project_id]);
+		}
+
+		#[test]
+		fn multiple_fields_edited() {
```

Comment:
Not sure if this test makes much sense as it is using the same logic as create_project for validation. Think just checking if setting the same metadata and some different metadata is possible. I don't think we care a lot about if we are able to change a single item, as we are not comparing the two different Metadata structures, right?

Reply:
fair point, but might be good to decouple the desired behavior with the actual implementation no?

Reply:
![CleanShot 2024-04-03 at 13 11 17@2x](https://github.com/Polimec/polimec-node/assets/54085674/3206b4b4-e736-482d-a25f-3144bbbea9db)
That's already the case no?